test/test_xla_sharding.py

```diff
@@ -32,12 +30,27 @@ def test_xla_sharded_tensor(self):
     assert t3.size() == t1.size(
     ), "Sharded output should return unpartitioned tensor size."
 
-    device_ids = np.array(range(num_devices))
-    device_assignment = list(device_ids.reshape(mesh_shape))
-    # TODO(yeounoh) sharding_spec is broken. The intention is to
-    # check if the returned sharding spec holds the correct device
-    # assignment as intended.
-    # assert device_assignment == t1_sharded.sharding_spec[0]
+    # TODO(yeounoh) Check if the returned sharding spec holds the correct device
+    # assignments.
+
+  def test_mark_sharding(self):
+    t1 = torch.randn(1, 128, device='cpu')
+    t2 = torch.randn(1, 128, device='cpu')
+    expected = t1 @ t2.T
+
+    xt1 = t1.to(xm.xla_device())
+    xt2 = t2.to(xm.xla_device())
+    n_devices = xm.xrt_world_size()
```

Comment:
This will be 1 or 2 depending on the setup. SPMD is not available on GPU. I think we should start a separate workstream to support GPU -- the device assignment et al would need more work.

Reply:
Added 
```
@unittest.skipIf(
      len(xm.get_xla_supported_devices("GPU")) > 0,
      "SPMD is currently not supported on GPU.")
      ```

Reply:
Created a bug to track GPU enablement.